samples/CommunityToolkit.Maui.Sample/Pages/Converters/IndexToArrayItemConverterPage.xaml

```diff
@@ -4,12 +4,60 @@
+    <pages:BasePage.BindingContext>
+        <vm:IndexToArrayItemConverterViewModel />
+    </pages:BasePage.BindingContext>
+    <ContentPage.Resources>
+        <mct:IndexToArrayItemConverter x:Key="IndexToArrayItemConverter" />
+        <x:Array x:Key="StringArray" Type="{x:Type x:String}">
+            <x:String>Value 0</x:String>
+            <x:String>Value 1</x:String>
+            <x:String>Value 2</x:String>
+            <x:String>Value 3</x:String>
+            <x:String>Value 4</x:String>
+            <x:String>Value 5</x:String>
+            <x:String>Value 6</x:String>
+            <x:String>Value 7</x:String>
+        </x:Array>
+    </ContentPage.Resources>
     <pages:BasePage.Content>
         <StackLayout>
-            <Label Text="IndexToArrayItemConverter"
-                VerticalOptions="CenterAndExpand" 
-                HorizontalOptions="CenterAndExpand" />
+            <Label Text="The IndexToArrayItemConverter is a converter that allows users to convert a int value binding to an item in an array. The int value being data bound represents the indexer used to access the array. The array is passed in through the ConverterParameter."
+                VerticalOptions="FillAndExpand" 
+                HorizontalOptions="FillAndExpand" />
+                
+                <StackLayout Margin="0,20,0,20">
+                    <Label Text="Change SelectedIndex to change SelectedItem" 
+                           FontAttributes="Italic"/>
+                    <StackLayout Orientation="Horizontal" 
+                                 Spacing="10">
+                    <Stepper x:Name="Stepper" 
+                             Minimum="0" 
+                             Maximum="7"
+                             Value="{Binding SelectedIndex}"
+                             Increment="1"/>
+                    <Label Text="{Binding Source={x:Reference Stepper},
+                                           Path=Value, StringFormat='Selected index: {0}'}"
+                           VerticalOptions="Center"
+                           FontAttributes="Bold"/>
+                    </StackLayout>
+                </StackLayout>
+
+            <Label Text="Change SelectedItem to change SelectedIndex" FontAttributes="Italic"/>
+            <CollectionView
+                SelectionMode="Single"
+                SelectedItem="{Binding SelectedIndex, Converter={StaticResource IndexToArrayItemConverter}, ConverterParameter={StaticResource StringArray}}"
+                ItemsSource="{StaticResource StringArray}">
+                <CollectionView.ItemTemplate>
+                    <DataTemplate>
+                        <ContentView>
+                            <Label Padding="5" Text="{Binding}"/>
+                        </ContentView>            
+                    </DataTemplate>
+                </CollectionView.ItemTemplate>
+            </CollectionView>
         </StackLayout>
```

Comment:
Let's use a Grid instead to flatten the UI layers and improve performance over nesting a `StackLayout` inside of a `StackLayout` inside of a `StackLayout`.
> ...don't attempt to reproduce the appearance of a specific layout by using combinations of other layouts, as this results in unnecessary layout calculations being performed. For example, don't attempt to reproduce a Grid layout by using a combination of StackLayout instances.

https://docs.microsoft.com/xamarin/xamarin-forms/deploy-test/performance#choose-the-correct-layout?WT.mc_id=mobile-0000-bramin

Reply:
Hi @brminnick, thanks for the feedback. I've replaced the StackLayouts with a Grid.